api/schemas/views.py

```diff
@@ -28,10 +30,18 @@ class RegistrationSchemaList(JSONAPIBaseView, generics.ListAPIView):
 
     ordering = ('-id',)
 
-    # overrides ListCreateAPIView
-    def get_queryset(self):
-        return RegistrationSchema.objects.filter(schema_version=LATEST_SCHEMA_VERSION, active=True)
+    def get_default_queryset(self):
+        if waffle.switch_is_active('filter_schemas_registration_on_active'):
+            return RegistrationSchema.objects.filter(schema_version=LATEST_SCHEMA_VERSION)
+        else:
+            return RegistrationSchema.objects.filter(schema_version=LATEST_SCHEMA_VERSION, active=True)
 
+    # overrides ListAPIView
+    def get_queryset(self):
```

Comment:
Any reason this function can't be:
```
def get_queryset(self):
	return self.get_queryset_from_request()
```
?

It seems fine to allow filtering by active when the switch if off (because it will essentially be a noop).

Reply:
This doesn't work because `get_queryset_from_request` is filtering the default queryset so filtering is still happening but only on the default queryset AKA only active schemas. Filtering on `?filter[active]=false` would brokenly returns no results.

Reply:
Hmm, I'm confused. When the switch is off, `?filter[active]=false` would return zero results, but that would be the correct functionality.

Reply:
When the switch is off, `?filter[active]=false` is ignored and returns only active results, that's the current behavior.

Reply:
I'm still not sure I understand why the switch check needs to be in this function and in `get_default_queryset()`. We don't need to maintain the current behavior where `?filter[active]=false` returns all active schemas because, as you said, that only happens because filtering isn't currently implemented and the param is ignored. 

AFAICT, given the following code:
```
def get_default_queryset(self):
	if waffle.switch_is_active(FILTER_REG_SCHEMAS_ON_ACTIVE):
		return RegistrationSchema.objects.filter(schema_version=LATEST_SCHEMA_VERSION)
	else:
		return RegistrationSchema.objects.filter(schema_version=LATEST_SCHEMA_VERSION, active=True)

def get_queryset(self):
	return self.get_queryset_from_request()
```

we would get the following for each case:
```
?filter[active]=true with switch ON: all active schemas (correct)
?filter[active]=false with switch ON: all inactive schemas (correct)
?filter[active]=true with switch OFF: all active schemas (correct)
?filter[active]=false with switch OFF: zero schemas (correct) 
```